Fix GEOSCHEMchem_ExtData.yaml for GEOS-Chem transport tracers in GEOS

lizziel opened the pull request:
### Name and Institution (Required)

Name: Lizzie Lundgren
Institution: Harvard University

### Describe the update

This PR contains fixes to allow running ExtData2G with GEOS-Chem transport tracer simulation within GEOS. Changes include:
- Update formatting of sample logicals (no quotes)
- Remove annual scale factor imports since used only with HTAP which is not used in GEOS
- Add valid date ranges for all imports that span multiple years
- Rename ExtData yaml file sampling and collection names to be both more generic and descriptive
- Update the Yuan condensed XLAI inventory to use the same source as the GEOS-Chem standard model
- Change the GEOS-IT data path prefix to the shared directory on discover
- Update the data interpolation used for XLAI to match what is used for the GEOS-Chem full chemistry simulation in GEOS

Note also that I added a section to the GEOS-Chem changelog for geos/develop changes. This can be used for all other updates that go into geos/develop. I can use that information when bringing GEOS updates into the standard model.

### Expected changes

Successfully read GEOSCHEMchem_ExtData.yaml when using GEOS-Chem transport tracer simulation in GEOSgcm.

### Reference(s)

None

### Related Github Issue

None

lizziel committed:
Fixed GEOSCHEMchem_ExtData.yaml for GEOS-Chem transport tracers in GEOS

Signed-off-by: Lizzie Lundgren <elundgren@seas.harvard.edu>

lizziel commented:
This is draft until I get the transport tracers simulation to successfully complete.

viral211 commented on run/GEOS/TransportTracers/GEOSCHEMchem_ExtData.yaml:
```diff
@@ -27,6 +27,8 @@ Collections:
   TT_EDGAR_v43.Seasonal.1x1.nc:
     template: /discover/nobackup/projects/gmao/share/dasilva/fvInput/ExtData/chemistry/EDGARv43/v2016-11/EDGAR_v43.Seasonal.1x1.nc
+  TT_GEOSCHEMchem_d5294_geosit_jan18.asm_const_0hr_glo_C180x180x6_slv.2018-01-01T0300Z.nc4:
+      template: /gpfsm/dnb06/projects/p171/dao_ops/archive/d5294_geosit_jan18/diag/Y2018/M01/d5294_geosit_jan18.asm_const_0hr_glo_C180x180x6_slv.2018-01-01T0300Z.nc4
```
hi @lizziel. Please use the alias `/discover/nobackup/projects/gmao/geos-it/` instead of `/gpfsm/dnb06/projects/p171/` in case the actual location of the folder changes

lizziel replied:
Will do

lizziel committed:
Add collection valid time ranges

Signed-off-by: Lizzie Lundgren <elundgren@seas.harvard.edu>

lizziel committed:
Rename collections and samplings to be more intuitive; fix LAI sampling

The new sampling definition for LAI matches what is used in the Extdata
yaml file for full chemistry. The samples also now specify extrapolation
as clim for all except constants. This means times outside the valid date
range of the data will use the closest year as climatology.

Signed-off-by: Lizzie Lundgren <elundgren@seas.harvard.edu>

lizziel committed:
Change collection name and path for GEOS-IT fields

This is a zero diff update

Signed-off-by: Lizzie Lundgren <elundgren@seas.harvard.edu>

lizziel committed:
Add valid date ranges for EDGAR SF6 and Yuan condensed XLAI

Signed-off-by: Lizzie Lundgren <elundgren@seas.harvard.edu>

lizziel committed:
Update data path for Yuan condensed XLAI to match standard GEOS-Chem

Signed-off-by: Lizzie Lundgren <elundgren@seas.harvard.edu>

lizziel committed:
Remove annual scale factors since only used with HTAP

Signed-off-by: Lizzie Lundgren <elundgren@seas.harvard.edu>

lizziel committed:
Fix incorrect import date ranges for several inventories

Signed-off-by: Lizzie Lundgren <elundgren@seas.harvard.edu>

lizziel committed:
Updated changelog

Signed-off-by: Lizzie Lundgren <elundgren@seas.harvard.edu>

lizziel commented:
These updates now enable ExtData2G to successfully run in GEOS using GEOS-Chem transport tracers.
